units/provisioning_create_host.go

```diff
@@ -119,7 +120,7 @@ func (j *createHostJob) Run(ctx context.Context) {
 		return
 	}
 
-	if j.host.ParentID == "" && !j.host.SpawnOptions.SpawnedByTask && !j.host.UserHost {
+	if j.host.IsSubjectToHostCreationThrottle() || j.host.Status == evergreen.HostBuilding {
```

Comment:
Why does this check if the host is in `evergreen.HostBuilding`?

Reply:
building hosts (largely containers,) have already started and this job does the next step in the process, so we want to continue to move those along, (also they don't eat away at the same resource pool as EC2 hosts, so it seems like a good compromise. 

Having said that, it looks like we should take it out.